unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapView.cs

```diff
@@ -27,9 +30,11 @@ public class NavmapView : MonoBehaviour
 
         private RectTransform RectTransform => rectTransform ??= transform as RectTransform;
         private BaseVariable<Transform> configureMapInFullscreenMenu => DataStore.i.exploreV2.configureMapInFullscreenMenu;
+        private NavmapSearchController navmapSearchController;
 
         private void Start()
         {
+            navmapSearchController = new NavmapSearchController(searchView, Environment.i.platform.serviceLocator.Get<IPlacesAPIService>(), new DefaultPlayerPrefs());
```

Comment:
This is a bit weird that View knows about Controller and also creates NavmapVisibilityBehaviour, which I was thinking is a composition root of the Map UI. But it is what it is... I actually put my logic inside NavmapVisibilityBehaviour (considering it a composition root). Maybe we both can isolate logic there and in case it could be extracted later and assembled properly. What do you think?

Reply:
My take on this is that when this will be refactored it will have a separate entry, this NavmapView is called a view but is acting a bit more like a controller, the next step refactor in my opinion would be to have a navmap controller creating everything, in any case I'd leave it like it is now to avoid spending time on it and if we manage to finish all the functionalities in time we can think of the proper refactor, maybe in the hardening week